File: Day9/Solution.go
package main
import "fmt"

var n int

func factorial(n int) int{
  if(n<=1){
    return 1
  } else{
    return n*factorial(n-1)
  }
}

func main(){
  fmt.Scan(&n)
  fmt.Print(factorial(n))
}

File: Day9/Solution.rs
use std::io;

fn main() {
    let mut input = String::new();
    match io::stdin().read_line(&mut input){
        Ok(_) => {
            match input.trim().parse::<i32>() {
                Ok(number) => {
                    println!("{0}", factorial(number));
                }
                Err(error) => {
                    panic!("{:?}", error);
                }
            }
        },
        Err(error) => {
            panic!("{:?}", error);
        }
    }
}

fn factorial(n: i32) -> i32{
    if n == 1 {
        return 1;
    }
    n * factorial(n-1)
}